skyvern/forge/sdk/forge_log.py

```
@@ -121,6 +122,154 @@ def __call__(self, logger: logging.Logger, name: str, event_dict: EventDict) ->
+
+
+def _generate_exception_hash(exc_type: type, tb: TracebackType) -> str:
+    """
+    Generate a stable hash for an exception based on:
+    - Exception type
+    - Stack trace (filename, line number, function name)
+
+    Excludes dynamic data like error messages to ensure the same
+    error from the same location always produces the same hash.
+    """
+    import hashlib  # noqa: PLC0415
```

Ruff (0.14.8)

Unused `noqa` directive (non-enabled: `PLC0415`)

Remove unused `noqa` directive

(RUF100)